python_modules/libraries/dagster-aws/dagster_aws/s3/sensor.py

```diff
@@ -22,19 +42,57 @@ def get_s3_keys(
         raise botocore.exceptions.ClientError("Failed to initialize s3 client")
 
     paginator = s3_session.get_paginator("list_objects_v2")
-    page_iterator = paginator.paginate(Bucket=bucket, Prefix=prefix)
+    page_iterator = paginator.paginate(Bucket=bucket, Prefix=prefix, Delimiter="")
 
-    contents = []
+    objects = []
     for page in page_iterator:
-        contents.extend(page.get("Contents", []))
+        contents = page.get("Contents", [])
+        objects.extend([Object(obj.get("Key"), obj.get("LastModified")) for obj in contents])
```

Comment:
can we just keep the whole dict? should be strictly better than limiting the information passed to the user

Reply:
@benpankow Pedram mentioned that you suggested using a named tuple. Would be curious to learn more as to the advantages of this vs. the whole object too.

Reply:
1. I'de recommend `TypedDict`. It's closer to the original dict returned by `boto3`. 
2. You can take a look at the `TypedDict` defined by `boto3-stubs` [here](https://youtype.github.io/boto3_stubs_docs/mypy_boto3_s3/type_defs/#objecttypedef) 

I think S3's API should be really stable so you could just copy this type annotation and plug it into our code.

Reply:
Interesting, those stubs are nice.

I see that currently boto3-stubs-lite is configured as `extras_require` > `stub` in our setup.py. Are there plans to make it included by default?

Copying the stubs feels not ideal, as the stub has some nesting, and would result in several `TypedDict` instances to maintain.

Reply:
@danielgafni - i've updated the PR to use TypedDicts. How does that look?

https://github.com/dagster-io/dagster/pull/25030/commits/27171c11c047e3d0921c8204b273ac42bd467566